packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -664,7 +664,15 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 				super.addTrackedListener(
 					event,
 					(msg: ISignalMessage | ISignalMessage[], documentId?: string) => {
-						if (!this.enableMultiplexing || !documentId || documentId === this.documentId) {
+						const targetClientId = Array.isArray(msg)
+							? msg[0].targetClientId
```

Comment:
I don't think we leave it that simple.
I would leave the multiplexing and docId checks first. Then msg will need to be filtered.
In similar cases we ensure we have an array with `const msgs = Array.isArray(msg) ? msg : [msg];`.
Then you can `.filter` for no targetClientId or a match.

Reply:
Changed to filter individual signal messages based on their targetClientId